File: python/anagram2.py
# Don't forget to run the tests (and create some of your own)

def anagrams_for(word, list_of_words):
	anagrams_found = []
	for this_word in list_of_words:
		test_this_word = list(this_word)
		for letter in word:
			if letter in test_this_word:
					test_this_word.remove(letter)
		if len(test_this_word) == 0:
				anagrams_found.append(this_word)
	return anagrams_found


#list_of_words = ["threads", "trashed", "hardest", "hatreds", "hounds"]
#print(anagrams_for("threads", list_of_words))

File: js/anagram2.js
exports.anagramsFor = function(word, listOfWords) {
    let anagramsFound = [];
    word = word.split('');
    listOfWords.forEach(thisWord => {
        if (thisWord.length == word.length) {
            var testThisWord = thisWord.split('');
            for (let i = 0; i < thisWord.split('').length; i++) {
                letter = thisWord[i];
                //console.log(letter)
                if (word.includes(letter)) {
                    testThisWord.splice(testThisWord.indexOf(letter), 1);
                }
            }
            if (testThisWord.length == 0) anagramsFound.push(thisWord);
        }
    })
    return anagramsFound
};